src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -69,145 +69,98 @@ internal bool TryGetNextHeader(Stream archiveStream, bool copyData)
-            {
-                _uid = Convert.ToInt32(paxEaUid);
-            }
-            if (gea.TryGetValue(PaxEaGid, out string? paxEaGid))
-            {
-                _gid = Convert.ToInt32(paxEaGid);
-            }
-            if (gea.TryGetValue(PaxEaUName, out string? paxEaUName))
-            {
-                _uName = paxEaUName;
-            }
-            if (gea.TryGetValue(PaxEaGName, out string? paxEaGName))
-            {
-                _gName = paxEaGName;
-            }
-        }
-
         // Reads the elements from the passed dictionary, which comes from the previous extended attributes entry,
         // and inserts or replaces those elements into the current header's dictionary.
         // If any of the dictionary entries use the name of a standard attribute, that attribute's value gets replaced with the one from the dictionary.
         // Unlike the historic header, numeric values in extended attributes are stored using decimal, not octal.
         // Throws if any conversion from string to the expected data type fails.
         internal void ReplaceNormalAttributesWithExtended(IEnumerable<KeyValuePair<string, string>> extendedAttributesEnumerable)
         {
-            Dictionary<string, string> ea = new Dictionary<string, string>(extendedAttributesEnumerable);
-            if (ea.Count == 0)
+            Debug.Assert(_extendedAttributes == null);
+            _extendedAttributes = new Dictionary<string, string>(extendedAttributesEnumerable);
+            if (_extendedAttributes.Count == 0)
             {
                 return;
             }
-            _extendedAttributes ??= new Dictionary<string, string>();
 
-            // First step: Insert or replace all the elements in the passed dictionary into the current header's dictionary.
-            foreach ((string key, string value) in ea)
-            {
-                _extendedAttributes[key] = value;
-            }
+            // Find all the extended attributes with known names and save them in the expected standard attribute.
 
-            // Second, find all the extended attributes with known names and save them in the expected standard attribute.
-            if (ea.TryGetValue(PaxEaName, out string? paxEaName))
+            // The 'name' header field only fits 100 bytes
+            if (_extendedAttributes.TryGetValue(PaxEaName, out string? paxEaName))
             {
                 _name = paxEaName;
             }
-            if (ea.TryGetValue(PaxEaLinkName, out string? paxEaLinkName))
+
+            // The 'linkName' header field only fits 100 bytes
+            if (_extendedAttributes.TryGetValue(PaxEaLinkName, out string? paxEaLinkName))
             {
                 _linkName = paxEaLinkName;
             }
-            if (ea.TryGetValue(PaxEaATime, out string? paxEaATime))
+
+            // PAX has no 'atime' header field, so it is always stored in the extended attributes
```

Comment:
Sorry for continuing to say basically the same thing over and over again... but... I don't know what this comment is trying to tell me.

PAX has no atime header.  Sure.  But we're assigning _aTime.  Are we sure we're not PAX?  If so, why are we talking about PAX?  What the code says, with no comment, is "If there's a PaxEaATime value, turn it into a DateTimeOffset and assign it to _aTime".  That seems pretty self-explanatory to me... turn an ATime value into an aTime value, but one is text and one is DateTimeOffset.

What's the comment for?  What's unintuitive? What does someone need to be careful about when they're making a change in the future?  If it's just restating what the code does, then the comment has zero to negative value, because it's distracting from reading the code, and will either get out of date or has to be updated.

Reply:
There is no public API that exposes AccessTime in PAX as a DateTimeOffset, so converting and storing in _atime is useless. I need to remove this if, and the one for ctime.

I need to double check if when writing a PAX entry, the atime and ctime are stored in the dictionary, if not there yet. If not there yet, it should acquire the value of UtcNow.